### PR TITLE
DBZ-2743 Unqouted IP address as part of username

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-2743

### CODE REVIEW

**Comment by jpechane** [comment]
Upstream PR - https://github.com/antlr/grammars-v4/pull/1947

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4 [hunk 1]
```diff
@@ -1220,6 +1220,11 @@ STRING_USER_NAME:                    (
                                      (
                                        SQUOTA_STRING | DQUOTA_STRING 
                                        | BQUOTA_STRING | ID_LITERAL
+                                       | IP_ADDRESS
```
Comment (rk3rn3r) [inline debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:1223]: IP_ADDRESS | IP6_ADDRESS ?

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4 [hunk 2]
```diff
@@ -1220,6 +1220,11 @@ STRING_USER_NAME:                    (
...
+                                     );
+IP_ADDRESS:                          (
+                                       [0-9]+ '.' [0-9.]+
+                                       | [0-9A-F:]+ ':' [0-9A-F:]+
```
Comment (rk3rn3r) [inline debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:1227]: Should we add IPv6 address pattern as well?

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4 (hunk 1 above)
Comment (gunnarmorling) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:1223]: Does MySQL support that?

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4 [hunk 3]
```diff
@@ -1220,6 +1220,11 @@ STRING_USER_NAME:                    (
                                      (
                                        SQUOTA_STRING | DQUOTA_STRING 
                                        | BQUOTA_STRING | ID_LITERAL
+                                       | IP_ADDRESS
+                                     );
+IP_ADDRESS:                          (
+                                       [0-9]+ '.' [0-9.]+
```
Comment (gunnarmorling) [inline debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:1226]: Isn't it always four numbers? I know it doesn't matter too much, but I think I'd give it simply as that: `[0-9]+ '.' + [0-9]+ '.' [0-9] + '.' + [0-9]`.

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4 (hunk 2 above)
Comment (gunnarmorling) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:1227]: I think that's this very line, no?

@jpechane, I'm confused though by th ":" within the first character class, is this intended? And wouldn't the second one match a trainling ":", too? Worth adding a test for sure.

Comment (gunnarmorling) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:1227]: Ah, and I guess this is not tolerant towards upper/lower-caseing?

Comment (gunnarmorling) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:1227]: I.e. I'd roughly have expected something like this: `[0-9A-Fa-f] (':' [0-9A-Fa-f])*`.

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4 (hunk 1 above)
Comment (rk3rn3r) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:1223]: yep. and #1977 has already lexer code for that. Maybe we close this one in favor of #1977?

Comment (gunnarmorling) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:1223]: In fact I was hoping we could go for this one, as it's much less complex.  Keep in mind we don't really have any interest in processing these statements, essentially it's good enough if the parser doesn't choke on them.

**Comment by gunnarmorling** [comment]
@jpechane, how to proceed, should we go with this one?

**Comment by gunnarmorling** [comment]
@jpechane, are you planning to wrap this up for tomorrow's release? Or rather CR1 next week?

**Comment by jpechane** [comment]
@gunnarmorling I've simplified the matching rules

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4 [hunk 4]
```diff
@@ -1220,6 +1220,11 @@ STRING_USER_NAME:                    (
                                      (
                                        SQUOTA_STRING | DQUOTA_STRING 
                                        | BQUOTA_STRING | ID_LITERAL
+                                       | IP_ADDRESS
+                                     );
+IP_ADDRESS:                          (
+                                       [0-9]+ '.' [0-9]+ '.' [0-9]+ '.' [0-9]+
+                                       | [0-9A-Fa-f:]+
```
Comment (gunnarmorling) [inline debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:1227]: I don't think that's correct, you wouldn't have a ":" at the last group, right?

**Review by gunnarmorling — APPROVED** [review]

**Pull request merged**